crates/ark/src/lsp/backend.rs

```diff
@@ -44,24 +48,68 @@ use crate::lsp::statement_range::StatementRangeParams;
 use crate::lsp::statement_range::StatementRangeResponse;
 use crate::r_task;
 
+// This enum is useful for two things. First it allows us to distinguish a
+// normal request failure from a crash. In the latter case we send a
+// notification to the client so the user knows the LSP has crashed.
+//
+// Once the LSP has crashed all requests respond with an error. This is a
+// workaround, ideally we'd properly shut down the LSP from the server. The
+// `Disabled` enum variant is an indicator of this state. We could have just
+// created an anyhow error passed through the `Result` variant but that would
+// flood the LSP logs with irrelevant backtraces.
```

Comment:
Maybe update this message to mention that we do force a shutdown? AFAICT this is us being defensive in case there are any other things happening after we've shutdown that will no longer be able to communicate with the client